src/html5-qrcode-scanner.ts

```diff
@@ -709,6 +737,30 @@ export class Html5QrcodeScanner {
         cameraActionStopButton.disabled = true;
         cameraActionContainer.appendChild(cameraActionStopButton);
 
+        // Optional torch button support.
+        const torchButton = TorchButton.create(
+            $this.html5Qrcode!,
+            {display: "none", marginLeft: "5px"},
+            // Callback in case of torch action failure.
+            (errorMessage) => {
+                $this.setHeaderMessage(
+                    errorMessage,
+                    Html5QrcodeScannerStatus.STATUS_WARNING);
+            }
+        );
+        const cameraActionTorchButton = torchButton.getTorchButton();
+        cameraActionContainer.appendChild(cameraActionTorchButton);
+
+        const showTorchButtonIfSupported = (settings: MediaTrackSettings) => {
+            // @ts-ignore
```

Comment:
Abstract API to test if torch button is supported into TorchButton class / file.